### PR TITLE
get rid of client-oauth2

### PR DESCRIPTION
**Pull request by huayang-codaio**
according to https://staging.coda.io/d/Quality-Tracker_d-GJF-DmEUK/My-Issues_sud-l#My-issues_tui9H/r20745&modal=true, client-oauth2 doesn't work for some oauth providers since it passes client secret via headers. 

this PR replaces client-oauth2 with direct requests to oauth providers (logic dup'ed from the packs-auth service). 

tested by 
- todoist pack
- fake_v2 pack with google auth. 
- force injecting a refresh and make sure it works.

### CODE REVIEW

File: testing/oauth_server.ts [hunk 1]
```diff
@@ -33,30 +40,57 @@ export function launchOAuthServerFlow({
...
-    redirectUri: makeRedirectUrl(port),
-    query: additionalParams,
+  const scope = requestedScopes ? requestedScopes.join(scopeDelimiter || ' ') : requestedScopes;
+  const redirectUri = makeRedirectUrl(port);
+  const callback = async (code: string): Promise<TokenCallbackResponse> => {
+    const params = {
+      grant_type: 'authorization_code',
+      code,
+      client_id: clientId,
+      client_secret: clientSecret,
```
Comment (ekoleda-codaio) [inline testing/oauth_server.ts:50]: It looks like this implementation always passes the `client_secret` as a POST form parameter, which doesn't match the production environment. There, it first attempts to pass it as a form parameter, and if that fails tries to pass it in an `Authorization: Basic` header. While long term we may want to make this an explicit option, for now we should likely replicate the logic in prod.

File: test/packs/fake_v2.ts [hunk 2]
```diff
@@ -2,6 +2,48 @@ import * as coda from '../..';
...
+  tokenUrl: 'https://accounts.google.com/o/oauth2/token',
+  scopes: ['https://www.googleapis.com/auth/calendar', 'profile', 'email'],
+  additionalParams: {
+    access_type: 'offline',
+    prompt: 'consent select_account',
+    session: false,
+  },
+
+  async getConnectionName(context) {
+    const response = await context.fetcher!.fetch({
```
Comment (jonathan-codaio) [inline test/packs/fake_v2.ts:19]: Super nit, don't need the `!`s anymore.

File: testing/oauth_server.ts (hunk 1 above)
Comment (jonathan-codaio) [reply testing/oauth_server.ts:50]: Yeah let's make sure we handle the fallback here.

Comment (huayang-codaio) [reply testing/oauth_server.ts:50]: sure we can do that.

**Review by jonathan-codaio — APPROVED** [review]

**Pull request merged**